features/perpetual/contract-state/GeneralInfo.tsx

```diff
@@ -139,19 +111,59 @@ function GeneralInfoView({
   );
 }
 
-// TODO: price feed will need to be updated to use uniswap and also be able to calculate funding rate
 const GeneralInfo = () => {
+  const { latestPrice, sourceUrls = [] } = PriceFeed.useContainer();
+  const { loading, error, data } = ContractState.useContainer();
+  const { activeSponsors = [] } = EmpSponsors.useContainer();
```

Comment:
I think for this component in particular, a new version will need to be made specifically for the perp.  So actually it should probably be removed for now. 

This is a really nuanced question though in general. Emp tools was not meant to support multiple versions or contract types, and many components are named / organized to show this.  Perp and EMP just happen to be similar enough that many of the components can be shared between them. So the question really is should everything be renamed to remove all references to EMP since technically its supporting emp and perp? I think technically yes but its like a real PITA